### DISCUSSION TITLE
calculate orientation while having positions

### DISCUSSION THREAD

**Discussion by eirtech** [stack-of-tasks/pinocchio#2068] [Q&A]
Hi, 

Thanks for all of your nice work.

If I know the positions of every joints of a robot arm (3D pose in the world coordinate) at a specific moment, is there  a simpler way to calculate each joints orientation rather than using IK?

ps. I also have the kinematic model of the arm.

Thanks in advanece

**Comment by stephane-caron** [comment]
If I understand correctly, you know the 3D position ${}\_W p^*\_{WJ}$ of each joint frame $J$ with respect to the inertial world frame $W$, and you search the configuration $q \in \mathcal{C}$ of the robot such that

```math
\forall J \in \mathit{joints}, \ {}_W p_{WJ}(q) = {}_W p^*_{WJ}
```

That's an IK problem (to be precise, an inverse geometry problem), it amounts to finding a reciprocal to the function $f : \mathcal{C} \to (\mathbb{R}^3)^n$ from configuration space to joint positions. To go further you can check out the discussion in, for instance, in this computer graphics paper: [Natural motion animation through constraining and deconstraining at will](https://doi.org/10.1109/TVCG.2003.1207443).

**Comment by eirtech** [comment]
Thanks for your answer. 
Yes, that was what I meant. I don't have the joints orientation and I just have the positions. The most important thing for me is to check if the positions are really feasible to reach for every joints. I was thinking if I could use the arm workspace as a limit to check if the positions are in a right area without using IK. But I didn't find proper source to do so.
